Tests/MongoSwiftTests/MongoDatabaseTests.swift

```diff
@@ -34,4 +34,34 @@ final class MongoDatabaseTests: MongoSwiftTestCase {
         expect(try db.runCommand(["asdfsadf": ObjectId()]))
                 .to(throwError(ServerError.commandError(code: 59, message: "", errorLabels: nil)))
     }
+
+    func testCreateCollection() throws {
+        let client = try MongoClient(MongoSwiftTestCase.connStr)
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
+
+        // test non-view options
+        let options = CreateCollectionOptions(
+            autoIndexId: true,
+            capped: true,
+            collation: ["locale": "fr"],
+            indexOptionDefaults: ["storageEngine": ["wiredTiger": ["configString": "access_pattern_hint=random"] as Document] as Document],
+            max: 1000,
+            size: 10000,
+            storageEngine: ["wiredTiger": ["configString": "access_pattern_hint=random"] as Document],
+            validationAction: "warn",
+            validationLevel: "moderate",
+            validator: ["phone": ["$type": "string"] as Document],
+            writeConcern: try WriteConcern(w: .majority)
+        )
+        expect(try db.createCollection("foo", options: options)).toNot(throwError())
+
+        // test view options
+        let viewOptions = CreateCollectionOptions(
```

Comment:
might be worth inserting a document to the original collection and then ensuring the pipeline is actually being used to make the view.

Reply:
maybe I should just run listCollections and look at the returned info? then we can validate everything has been correctly set.

Reply:
not sure if that might be brittle and vary across server versions? thoughts @mbroadst ?

Reply:
I don't think you need to worry about brittleness, we can skip sections of tests based on server version already.

Reply:
I think a test either way is fine. We generally just have to ensure we're passing the right stuff to libmongoc and can rely on their testbed for anything further than that.

Reply:
I ended up doing as I proposed with testing the returned collection info. `CreateCollectionOptions` is now `Decodable` so that we can directly parse the options document returned by `listCollections` back into options and verify they match the ones that we sent.